extensions/sql-migration/src/wizard/databaseSelectorPage.ts

```diff
@@ -0,0 +1,312 @@
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { MigrationWizardPage } from '../models/migrationWizardPage';
+import { MigrationStateModel, StateChangeEvent } from '../models/stateMachine';
+import * as constants from '../constants/strings';
+import { IconPath, IconPathHelper } from '../constants/iconPathHelper';
+import { debounce } from '../api/utils';
+
+const headerLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+	'border-bottom': '1px solid'
+};
+
+const styleLeft: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'left',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+};
+
+const styleCenter: azdata.CssStyles = {
+	'border': 'none',
+	'text-align': 'center',
+	'white-space': 'nowrap',
+	'text-overflow': 'ellipsis',
+	'overflow': 'hidden',
+};
+
+export class DatabaseSelectorPage extends MigrationWizardPage {
+	private _view!: azdata.ModelView;
+	private _databaseSelectorTable!: azdata.DeclarativeTableComponent;
+	private _dbNames!: string[];
+	private _dbCount!: azdata.TextComponent;
+	private _databaseTableValues!: azdata.DeclarativeTableCellValue[][];
+
+	constructor(wizard: azdata.window.Wizard, migrationStateModel: MigrationStateModel) {
+		super(wizard, azdata.window.createWizardPage(constants.SOURCE_CONFIGURATION, 'MigrationModePage'), migrationStateModel);
+	}
+
+	protected async registerContent(view: azdata.ModelView): Promise<void> {
+		this._view = view;
+
+		const flex = view.modelBuilder.flexContainer().withLayout({
+			flexFlow: 'row',
+			height: '100%',
+			width: '100%'
+		}).component();
+		flex.addItem(await this.createRootContainer(view), { flex: '1 1 auto' });
+
+		await view.initializeModel(flex);
+	}
+
+	public async onPageEnter(): Promise<void> {
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			return true;
+		});
+	}
+	public async onPageLeave(): Promise<void> {
+		this.migrationStateModel._databaseAssessment = this.selectedDbs();
+		this.wizard.registerNavigationValidator((pageChangeInfo) => {
+			return true;
+		});
+	}
+
+	protected async handleStateChange(e: StateChangeEvent): Promise<void> {
+	}
+
+
+	private createSearchComponent(): azdata.DivContainer {
+		let resourceSearchBox = this._view.modelBuilder.inputBox().withProps({
+			stopEnterPropagation: true,
+			placeHolder: constants.SEARCH,
+			width: 200
+		}).component();
+
+		resourceSearchBox.onTextChanged(value => this._filterTableList(value));
+
+		const searchContainer = this._view.modelBuilder.divContainer().withItems([resourceSearchBox]).withProps({
+			CSSStyles: {
+				'width': '200px',
+				'margin': '10px 8px 0px 0px'
+			}
+		}).component();
+
+		return searchContainer;
+	}
+
+	@debounce(500)
```

Comment:
What is the purpose of this line?

Reply:
it throttles the execution of the "decorated" method to only fire after no activity for 500 ms... preventing the search from updating if the user types quickly